### PR TITLE
07 vote article

### PR DESCRIPTION
**Pull request by smlbrd**

### CODE REVIEW

File: src/components/VoteHandler.jsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+function VoteHandler({ article_id, votes }) {
+  const [isClicked, setIsClicked] = useState(false);
+  const [voteAdded, setVoteAdded] = useState(0);
+  const [error, setError] = useState(null);
+
+  {
+    /* TODO: Each article can be liked ONCE PER PAGE VISIT
+    Implement user context later to make it one vote per user
+     */
+  }
```
Comment (nc-mohammed-dastgir) [inline src/components/VoteHandler.jsx:13]: This seems like a good idea, but context does not persist if a page is refreshed so you may need to consider an alternative to context for this behaviour

File: src/components/VoteHandler.jsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        setError('Something went wrong. Please try again later!');
+        setVoteAdded((currentVotesAdded) => {
+          return currentVotesAdded - 1;
+        });
+      });
+      setVoteAdded((currentVotesAdded) => {
+        setIsClicked(!isClicked);
+        return currentVotesAdded + 1;
+      });
+    }
```
Comment (nc-mohammed-dastgir) [inline src/components/VoteHandler.jsx:39]: There seems to be a lot of repeated code here, can you think of a way to refactor this to be more dry?

File: src/components/VoteHandler.jsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  }
+
+  if (error) return <p>{error}</p>;
```
Comment (nc-mohammed-dastgir) [inline src/components/VoteHandler.jsx:42]: I really like that error handling was considered, but this removes the like button preventing me from trying again unless I refresh

File: src/components/VoteHandler.jsx (hunk 1 above)
Comment (smlbrd) [reply src/components/VoteHandler.jsx:13]: That's heartbreaking, thanks for letting me know! :joy:  I'll do some research :saluting_face:

File: src/components/VoteHandler.jsx (hunk 2 above)
Comment (smlbrd) [reply src/components/VoteHandler.jsx:39]: Thanks for pointing me back at this, I think I'd scrambled my own brain a bit trying to implement it!

I've updated this to just use the first half of the if/else, and set the adjustment value using a ternary operator:

`const voteAdjustment = !isClicked ? 1 : -1;`

File: src/components/VoteHandler.jsx (hunk 3 above)
Comment (smlbrd) [reply src/components/VoteHandler.jsx:42]: I've moved `{error && <p>{error}</p>}` down to the bottom of the rendered component, so an error message will appear underneath the button without removing it.

**Pull request merged**